merino-settings/src/providers.rs

```diff
@@ -215,6 +217,60 @@ impl Default for StealthConfig {
     }
 }
 
+/// Settings for Merino suggestion providers.
+#[derive(Serialize, Deserialize)]
+pub struct SuggestionProviderSettings {
+    pub suggestion_providers: HashMap<String, SuggestionProviderConfig>,
```

Comment:
This requires each config file to have a top-level "suggestion_providers", which seems redundant. I've tried defining `struct SuggestionProviderSettings(pub HashMap<String, SuggestionProviderConfig>)`, but couldn't get the deserialization working.

Reply:
I'm not having trouble with tuple-types wrapping hash maps. Can you recreate the problem in a smaller case?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9d374e3390338ae95239835c11811ca5

Reply:
Yeah, that worked for me too. But when I changed this to

```rs
#[derive(Serialize, Deserialize)]
pub struct SuggestionProviderSettings (
    pub HashMap<String, SuggestionProviderConfig>
);
```

Setting `config/providers/test.yaml` to `{}` or a valid object. It yelled at me with this:

```
'Could not convert settings for suggestion providers: invalid type: map, expected tuple struct SuggestionProviderSettings', merino-settings/src/providers.rs:270:14
```

Reply:
Turns out I have to explicitly enable the [`transparent`](https://serde.rs/container-attrs.html#transparent) attribution to make it work.